src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -414,6 +414,13 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
 
         group = groupOption.getOrElse(throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"$groupId not found")))
 
+        groupOption2 <- groupId match {
+          case basicGroupName: WorkbenchGroupName => directoryDAO.loadGroup(basicGroupName)
+          case rpn: ResourceAndPolicyName => accessPolicyDAO.loadPolicy(rpn)
+        }
+
+        group2 = groupOption2.getOrElse(throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"$groupId not found")))
```

Comment:
you can do `group2 <- IO.fromEither(groupOption2.toRight(throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"$groupId not found")))).unsafeToFuture` (probably will be easier to move things to IO later this way too)